book/guide/quic.md

```diff
@@ -0,0 +1,210 @@
+    Signature Algorithm: Ed25519
+        Issuer: CN=Solana node
+        Validity
+            Not Before: Jan  1 00:00:00 1970 GMT
+            Not After : Jan  1 00:00:00 4096 GMT
+        Subject:
+        Subject Public Key Info:
+            Public Key Algorithm: Ed25519
+            [ PUBLIC KEY GOES HERE ]
+        X509v3 extensions:
+            X509v3 Subject Alternative Name: critical
+                DNS:localhost
+            X509v3 Basic Constraints: critical
+                CA:FALSE
+    Signature Algorithm: Ed25519
+         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
+         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
+         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:ff:
+         ff:ff:ff:ff:ff:ff:ff:ff:ff:ff
+```
+
+</details>
+
+<details>
+<summary>Fake certificate hex dump</summary>
+
+To construct a fake certificate, copy paste the Ed25519 public key at
+offset 0x64.
+
+```
+00000000: 3081 f630 81a9 a003 0201 0202 0801 0101  0..0............
+00000010: 0101 0101 0130 0506 032b 6570 3016 3114  .....0...+ep0.1.
+00000020: 3012 0603 5504 030c 0b53 6f6c 616e 6120  0...U....Solana
+00000030: 6e6f 6465 3020 170d 3730 3031 3031 3030  node0 ..70010100
+00000040: 3030 3030 5a18 0f34 3039 3630 3130 3130  0000Z..409601010
+00000050: 3030 3030 305a 3000 302a 3005 0603 2b65  00000Z0.0*0...+e
+00000060: 7003 2100 ffff ffff ffff ffff ffff ffff  p.!.............
+00000070: ffff ffff ffff ffff ffff ffff ffff ffff  ................
+00000080: ffff ffff a329 3027 3017 0603 551d 1101  .....)0'0...U...
+00000090: 01ff 040d 300b 8209 6c6f 6361 6c68 6f73  ....0...localhos
+000000a0: 7430 0c06 0355 1d13 0101 ff04 0230 0030  t0...U.......0.0
+000000b0: 0506 032b 6570 0341 00ff ffff ffff ffff  ...+ep.A........
+000000c0: ffff ffff ffff ffff ffff ffff ffff ffff  ................
+000000d0: ffff ffff ffff ffff ffff ffff ffff ffff  ................
+000000e0: ffff ffff ffff ffff ffff ffff ffff ffff  ................
+000000f0: ffff ffff ffff ffff ff                   .........
+```
+
+</details>
+
+### Raw Public Keys
+
+Raw Public Keys are an alternative certificate type specified in
+[RFC 7250](https://datatracker.ietf.org/doc/html/rfc7250).
+
+fd_quic/fd_tls support Ed25519 raw public keys.
+
+<details>
+<summary>Raw Public Key ASN.1 dump</summary>
+
+```
+    0:d=0  hl=2 l=  42 cons: SEQUENCE
+    2:d=1  hl=2 l=   5 cons: SEQUENCE
+    4:d=2  hl=2 l=   3 prim: OBJECT            :Ed25519
+    9:d=1  hl=2 l=  33 prim: BIT STRING
+```
+
+</details>
+
+<details>
+<summary>Raw Public Key Hexdump</summary>
+
+To construct a raw public key, paste the Ed25519 public key at offset 0x0c.
+
+
+```
+00000000: 302a 3005 0603 2b65 7003 2100 ffff ffff  0*0...+ep.!.....
+00000010: ffff ffff ffff ffff ffff ffff ffff ffff  ................
+00000020: ffff ffff ffff ffff ffff ffff            ............
+```
+
+</details>
+
+### Next Protocol Selection
+
+fd_quic/fd_tls uses the ALPN TLS extension (RFC 7301) to verify that
```

Comment:
Sometimes "fd_quic/fd_tls" is used as singular, other times as plural. Either one is fine, but keep it consistent.